src/components/RadioButton/RadioButton.jsx

```diff
@@ -53,7 +46,7 @@ const RadioButton = forwardRef(
             value={value}
             name={name}
             disabled={disabled}
-            {...checkedProps}
+            defaultChecked={checked || defaultChecked}
```

Comment:
but checked prop is for controlled state. it's means that we want that when clicking on the radiobutton it's state will not changed (only if we changed it by our prop) - and not changed if we click once / twice.